internal/driver/webhtml.go

```diff
@@ -767,125 +767,94 @@ function viewer(baseUrl, nodes) {
-    ["PB", 2**50],
-    ["ns", 1e-9],
-    ["us", 1e-6],
-    ["ms", 1e-3],
-    ["s", 1],
-    ["mins", 60.0],
-    ["hrs", 3600.0],
-    ["days", 3600.0 * 24.0],
-    ["wks", 3600.0 * 24.0 * 7.0],
-    ["yrs", 3600.0 * 24.0 * 7.0 * 365.0],  // Ignore leap years like measurement.go
-  ])
+function makeTopTable(total, entries) {
+  const rows = document.getElementById("rows")
+  if (rows == null) return
 
-  // Which column are we currently sorted by and in what order?
-  let currentColumn = 0
-  let descending = true
+  // Store initial index in each entry so we have stable node ids for selection.
+  for (let i = 0; i < entries.length; i++) {
+    entries[i].Id = "node" + i
+  }
 
-  function sortBy(column, numeric) {
-    const rows = document.getElementById("rows")
-    if (rows == null) return
+  // Which column are we currently sorted by and in what order?
+  let currentColumn = ""
+  let descending = false
+  sortBy("Flat")
 
+  function sortBy(column) {
     // Update sort criteria
     if (column == currentColumn) {
       descending = !descending  // Reverse order
     } else {
       currentColumn = column
-      descending = (column != 5)  // Name is ascending by default
-    }
-
-    // Get info for all rows we want to sort.
-    let list = []
-    for (const r of rows.children) {
-      if (r.children.length <= column) continue
-      const c = r.children[column]
-      const val = c.innerText
-
-      // Extract numeric and string portions from val.
-      let num = 0
-      let suffix = val
-      if (numeric) {
-        const match = val.match(numParseRe)
-        if (match != null) {
-          const n = parseFloat(match[1])
-          if (!isNaN(n)) {
-            num = n
-            suffix = match[3]
-            if (units.has(suffix)) {
-              num *= units.get(suffix)
-              suffix = ""
-            }
-          }
-        }
-      }
-      list.push([num, suffix, r])
+      descending = (column != "Name")
     }
 
+    // Sort according to current criteria.
     function cmp(a, b) {
-      if (a[0] < b[0]) return -1
-      if (a[0] > b[0]) return +1
-      if (a[1] < b[1]) return -1
-      if (a[1] > b[1]) return +1
+      const av = a[currentColumn]
+      const bv = b[currentColumn]
+      if (av < bv) return -1
+      if (av > bv) return +1
       return 0
     }
-    function dcmp(a, b) { return -cmp(a, b) }
-
-    list.sort(descending ? dcmp : cmp)
-
-    // Sum% column holds the running sum of rows and since we
-    // have re-ordered rows, recompute that.
-    let sumPercent = 0.0
-    for (const r of list) {
-      const row = r[2]
-      if (row.children.length < 3) continue
-      const flatPercent = parseFloat(row.children[1].innerText)
-      if (!isNaN(flatPercent)) sumPercent += flatPercent
-      row.children[2].innerText = sumPercent.toFixed(2) + "%"
+    entries.sort(cmp)
+    if (descending) entries.reverse()
+
+    function addCell(tr, val) {
+      const td = document.createElement('td')
```

Comment:
Should all JS code here have semicolons at the end of statements? Seems like it is a [recommended practice](https://stackoverflow.com/questions/444080/do-you-recommend-using-semicolons-after-every-statement-in-javascript). This isn't related to this PR, just curious.

Reply:
Google Javascript style guide, which we might as well follow, says to use semicolons.

https://google.github.io/styleguide/jsguide.html#formatting-semicolons-are-required

However we should not extend this PR to add semicolons and should make a distinct change.  And I am worried about churning the code if there are substantial other changes in flight out there.  The only potential one is the flamegraph one, but that seems to not be progressing?

Reply:
Agree that adding semicolons (or any kind of wide formatting changes) would be a separate CL. The flamegraph PR #188 seems to have stalled for some reason, yes. I'll ping that. We can defer adding semicolons until that PR is done. But it doesn't seem that PR makes a lot of changes to the existing JS code?

Reply:
I agree, I think I would capture potential problems with automatic semicolon insertion (ASI) in the code here, but probably it would be worth considering for a future PR extracting the static JS into a separate lintable file (that usually demands using semicolons anyway).